### PR TITLE
Chore: extract kube-api from core

### PR DESCRIPTION
**Pull request by jakolehm**

### CODE REVIEW

File: packages/core/src/common/k8s-api/api-manager/manager.injectable.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import { storesAndApisCanBeCreatedInjectionToken } from "../stores-apis-can-be-c
 import { ApiManager } from "./api-manager";
 import { computedInjectManyInjectable } from "@ogre-tools/injectable-extension-for-mobx";
 import { kubeObjectStoreInjectionToken } from "./kube-object-store-token";
-import { kubeApiInjectionToken } from "../kube-api/kube-api-injection-token";
+import { kubeApiInjectionToken } from "@k8slens/kube-api";
```
Comment (Nokel81) [inline packages/core/src/common/k8s-api/api-manager/manager.injectable.ts]: Oh so you decided to have the injection token be part of this package?

File: packages/utility-features/kube-api/package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  "main": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "author": {
+    "name": "OpenLens Authors",
+    "email": "info@k8slens.dev"
+  },
+  "license": "MIT",
+  "homepage": "https://github.com/lensapp/lens",
+  "scripts": {
+    "build": "webpack",
```
Comment (samitiilikainen) [inline packages/utility-features/kube-api/package.json]: Should use `lens-webpack-build` instead?

Comment (Nokel81) [reply packages/utility-features/kube-api/package.json]: probably

File: packages/utility-features/kube-api/src/kube-api-parse.test.ts [hunk 3]
```diff
@@ -103,129 +131,16 @@ const tests: KubeApiParseTestData[] = [
...
-  "/apis/apiextensions.k8s.io/v1beta1//prometheuses.monitoring.coreos.com",
-  "/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/",
-
-  "//v1beta1/customresourcedefinitions/prometheuses.monitoring.coreos.com",
-  "/api//v1beta1/customresourcedefinitions/prometheuses.monitoring.coreos.com",
-  "/api//customresourcedefinitions/prometheuses.monitoring.coreos.com",
-  "/api/v1beta1//prometheuses.monitoring.coreos.com",
-  "/api/v1beta1/customresourcedefinitions/",
-];
+const invalidTests = [undefined, "", "ajklsmh"];
```
Comment (Iku-turso) [inline packages/utility-features/kube-api/src/kube-api-parse.test.ts]: Here `tests` and `invalidTests` have degenerated to fewer scenarios. The commit name doesn't hint the motivation behind this. The missing scenarios describe an actual bug that was in production.

**Review by Nokel81 — DISMISSED** [review]

**Review by Nokel81 — DISMISSED** [review]

**Review by Nokel81 — APPROVED** [review]

**Pull request merged**